websites/Z/ZARP Gaming/presence.ts

```diff
@@ -0,0 +1,119 @@
+					break;
+				case "/index.php/profile/profile-edit":
+					presenceData.details = "Editing profile";
+					break;
+				case "/index.php/2013-11-13-21-35-14/":
+					presenceData.details = "Viewing a section:";
+					presenceData.state = document.querySelector<HTMLElement>(
+						"#Kunena > div.kblock.kpathway.breadcrumbs-1 > div > div > div > div.path-element > a"
+					).textContent;
+					presenceData.buttons = [
+						{
+							label: "Visit Section",
+							url: window.location.href,
+						},
+					];
+					break;
+				case "/index.php/new-topic":
+					presenceData.details = "Creating a topic in section:";
+
+					for (const option of document.querySelector<HTMLSelectElement>(
+						"#postcatid"
+					)) {
+						if (option.selected) {
+							presenceData.state = option.textContent
+								.replace(/^[-\s]+/, "")
+								.replace(/[-\s]+/g, " ");
+							break;
+						}
+					}
+					break;
+			}
+
+			switch (true) {
+				case window.location.pathname.includes("/forum/profile/"):
+					presenceData.details = "Viewing a profile:";
+					presenceData.state = document.title.replace("Profile for ", "");
+					presenceData.buttons = [
+						{
+							label: "Visit Profile",
+							url: window.location.href,
+						},
+					];
+					break;
+				case window.location.pathname.includes("/topic/create"):
+					presenceData.details = "Creating a topic in section:";
+					presenceData.state = document.querySelector<HTMLElement>(
+						"#Kunena > div.kblock.kpathway.breadcrumbs-1 > div > div > div > div:nth-child(3) > a"
+					).textContent;
+					break;
+				case window.location.pathname.endsWith("/reply"):
+					presenceData.details = "Replying to a topic:";
+					presenceData.state =
+						document.querySelector<HTMLInputElement>("#subject").value;
+					break;
+				case window.location.pathname.includes("/faq/"):
+					presenceData.details = "Viewing FAQ:";
+					presenceData.state = document.title;
+					break;
+				case window.location.pathname.includes(
+					"/index.php/forum/announcement/"
+				):
+					if (window.location.pathname.includes("list"))
+						presenceData.details = "Viewing the announcement list";
+					else if (window.location.pathname.includes("edit")) {
+						presenceData.details = "Editing an announcement:";
+						presenceData.state = document.querySelector<HTMLInputElement>(
+							"#kannouncement > div.kbody > div > form > div:nth-child(4) > label:nth-child(1) > input"
+						).value;
+					} else {
+						presenceData.details = "Viewing an announcement:";
+						presenceData.state = document.querySelector<HTMLElement>(
+							"#Kunena > div.kblock.kannouncement > div.kheader > h2 > span"
+						).textContent;
+					}
+					break;
+			}
+
+			if (!presenceData.details) {
+				presenceData.details = "Viewing a page:";
+				presenceData.state = document.title;
+			}
+			break;
+		case "control.zarpgaming.com":
+			presenceData.details = "Server Control Panel";
+			break;
```

Comment:
is that all? Can't you show more?

Reply:
Nope, due to privacy content on this page cannot be shown regardless, though still showing you are on the control panel is nice to have for us.

Reply:
You can have a setting for that